frontend/cypress/integration/common/hooks.ts

```diff
@@ -85,3 +88,17 @@ Before({ tags: '@error-rates-app' }, async function () {
 Before({ tags: '@sleep-app' }, async function () {
   install_demoapp('sleep');
 });
+
```

Comment:
@nrfox how do we install Istio CRD's in multicluster setup? Do we want to implement logic below in cypress only? Just thinking about having extra script to do that for better re usability. (non blocking)

Reply:
For primary-remote, CRDs are installed on the primary cluster but not the remote cluster like in what you see here: https://istio.io/latest/docs/setup/install/multicluster/primary-remote/. "config" in the diagram means the istio CRDs.

The primary-remote installation happens in the kiali hack scripts that run before any of tests to setup the environment. Here's where the remote cluster gets setup: https://github.com/kiali/kiali/blob/master/hack/istio/multicluster/install-primary-remote.sh#L88. The `--profile=remote` flag is what installs the necessary components for the remote cluster but not the CRDs.

In general the Kiali e2e and cypress tests right now assume that the environment is all setup ahead of time by the hack scripts before they run but sometimes they do things like check that the necessary demo apps are installed iirc. I think we should keep the environment setup outside the scope of the tests so that the tests can be run against a variety of environments (locally on kind or against a cloud env etc.) but installing istio itself could be part of the test framework since that will probably be environment agnostic and it would be nice to ensure things about the istio installation before running specific tests like for multi cluster or testing against multiple revisions. That's a larger change though as today installing istio is part of the environment setup.

Reply:
@ScriptingShrimp @matejnesuta ^^